packages/compiler/core/checker.ts

```diff
@@ -334,13 +430,15 @@ export function createChecker(program: Program): Checker {
       return templateInstantiation[index];
     }
 
-    return createType({
+    return createAndFinishType({
       kind: "TemplateParameter",
       node: node,
     });
   }
 
-  function checkTypeReference(node: TypeReferenceNode): Type {
+  function checkTypeReference(
+    node: TypeReferenceNode | MemberExpressionNode | IdentifierNode
```

Comment:
Not sure the context for this yet, but one thing to note is that a MemberExpression or Identifier that references a type won't be completed in IDE if it's not parented by TypeReferenceNode. Decorators are already special cased for this.

Reply:
In this case, I wanted to call `checkTypeReference` on the projection selector which doesn't allow selecting an instantiated type so I don't parse a TypeReference there. I suppose we could allow that, or else throw an error if I find arguments I suppose. I can change this.

But in general completion will likely need more special cases. The general the issue is that projections have a slightly different grammar. Outside of projections, member expressions are a primary expression. Inside projections, we have call expressions which can participate in member expressions (e.g. `foo().bar.baz()`) and we don't (presently) support instantiating templates. If we did, we'd do them differently because those too could participate in call and member expressions (`Foo<Bar>.properties.forEach()`).

Reply:
You can now project any reference expression.